ui/src/app/shared/components/orb/policy/policy-groups/policy-groups.component.ts

```diff
@@ -29,28 +32,40 @@ export class PolicyGroupsComponent implements OnInit, OnDestroy {
 
   errors;
 
-  constructor(protected datasetService: DatasetPoliciesService,
-              protected groupService: AgentGroupsService,
-              protected dialogService: NbDialogService,
-              protected router: Router,
-              protected route: ActivatedRoute) {
+  constructor(
+    protected datasetService: DatasetPoliciesService,
+    protected groupService: AgentGroupsService,
+    protected dialogService: NbDialogService,
+    protected router: Router,
+    protected route: ActivatedRoute,
+  ) {
+    this.refreshPolicy = new EventEmitter<string>();
     this.policy = {};
     this.datasets = [];
     this.groups = [];
     this.errors = {};
   }
 
   ngOnInit(): void {
-    this.subscription = this.retrievePolicyDatasets()
-      .pipe(mergeMap(datasets => this.retrieveAgentGroups(datasets))).subscribe(resp => {
-        this.datasets = resp;
 
-        if (!this.datasets || this.datasets.length === 0) {
-          this.errors['nogroup'] = 'This policy is not applied to any group.';
-        }
+  }
 
-        this.isLoading = false;
-      });
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.policy) {
+      if (this.subscription) {
+        this.subscription?.unsubscribe();
+      }
+      this.subscription = this.retrievePolicyDatasets()
+        .pipe(mergeMap(datasets => this.retrieveAgentGroups(datasets))).subscribe(resp => {
+          this.datasets = resp;
+
+          if (!this.datasets || this.datasets.length === 0) {
+            this.errors['nogroup'] = 'This policy is not applied to any group.';
+          }
```

Comment:
Aren't we using any lib for these checks? Like lodash.isEmpty()?

Reply:
Not directly, I think.
Perhaps it is used by some dependency.